### PR TITLE
feat(hub-common): add additional resources to IHubContent composition logic

### PR DESCRIPTION
**Pull request by sonofflynn89**
affects: @esri/hub-common

1. Description:
Part of [3304](https://devtopia.esri.com/dc/hub/issues/3304)
Added to the IHubContent interface and created additional helpers to parse and transform additional resources from formal item metadata

1. [x] ran commit script (`npm run c`)

### CODE REVIEW

File: packages/common/src/content/_internal.ts [hunk 1]
```diff
@@ -11,19 +11,20 @@
 import {
   ILayerDefinition,
   IFeatureServiceDefinition,
+  parseServiceUrl,
 } from "@esri/arcgis-rest-feature-layer";
 import { IItem } from "@esri/arcgis-rest-portal";
 import { ISpatialReference } from "@esri/arcgis-rest-types";
 import { IHubContent } from "../core";
 import {
-  BBox,
```
Comment (sonofflynn89) [inline packages/common/src/content/_internal.ts:19]: Unused import

File: packages/common/src/content/index.ts [hunk 2]
```diff
@@ -12,8 +12,6 @@ import { getServiceTypeFromUrl } from "../urls";
 import { getHubRelativeUrl, isPageType } from "./_internal";
 import { camelize } from "../util";
 import {
-  getLayerIdFromUrl,
-  isFeatureService,
```
Comment (sonofflynn89) [inline packages/common/src/content/index.ts:16]: More unused imports

File: packages/common/test/compose.test.ts [hunk 3]
```diff
@@ -196,83 +196,123 @@ describe("composeContent", () => {
         );
       });
     });
-  });
```
Comment (sonofflynn89) [inline packages/common/test/compose.test.ts:199]: The changes on this file look like a lot, but if you use the vs code view (press `.`), you'll see that I moved a parenthesis/bracket pair. Turns out that `publishedDate` tests were just outside of the describe that defined `item`, so undefined was being passed in.

**Review by tomwayson — APPROVED** [review]
well done!

**Pull request merged**